src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -30,6 +35,10 @@ internal sealed class MsQuicConnection : QuicConnectionProvider
         private IPEndPoint? _localEndPoint;
         private readonly EndPoint _remoteEndPoint;
         private SslApplicationProtocol _negotiatedAlpnProtocol;
+        private bool _isServer;
```

Comment:
Could any of these fields be readonly?
Do we need both: `_isServer` and `_remoteCertificateRequired`? From the ctors it looks like they're just opposites.

Reply:
for client, _remoteCertificateRequired is always true. For server this is optional depending on setting. That part is not wired completely as client certs are not fully functional.